src/sidebar/components/annotation.js

```diff
@@ -197,6 +198,12 @@ function AnnotationController(
       return;
     }
 
+    if (!self.annotation.user) {
+      // make the bug https://github.com/hypothesis/client/issues/12 explicit to user so they can at least log in
+      bridge.call('showSidebar');
+      flash.error('Sorry, drafts are currently broken. Please log in first.');
```

Comment:
As far as the location of this compared to the annotator file. I think this belongs here since the annotator doesn't know about the user being logged in or not. This should be moved down after the [if not highlight check](https://github.com/hypothesis/client/blob/546b07716ed1977f580c9585762dd4900c740601/src/sidebar/components/annotation.js#L207 ) though since we only want to do this if it is a highlight and the user is logged out.